src/midi/midi_alsa.cpp

```diff
@@ -215,14 +232,17 @@ static alsa_address find_seq_input_port()
 	// but only first two ones generate sound (even though all 4
 	// ones are marked as writable).
 	//
-	auto find_input_port = [&seq_addr](auto *client_info, auto *port_info) {
+	auto find_input_port = [&pattern, &seq_addr](auto *client_info,
+	                                             auto *port_info) {
 		const auto *addr = snd_seq_port_info_get_addr(port_info);
 		const auto caps = snd_seq_port_info_get_capability(port_info);
 		const auto client_type = snd_seq_client_info_get_type(client_info);
 
 		const bool is_user_client = (client_type == SND_SEQ_USER_CLIENT);
 		const bool is_new_client = (addr->client != seq_addr.client);
-		if (is_new_client && is_user_client && port_is_writable(caps)) {
+		const bool match = port_name_matches(pattern, client_info, port_info);
+		const bool is_candidate = (pattern.empty() ? is_user_client : match);
```

Comment:
`const bool is_candidate = pattern.empty() ? is_user_client : port_name_matches(pattern, client_info, port_info);`

Just an alternative with the benefit of only running the `port_name_matches(` if needed.

Reply:
I formatted this piece of code a bit differently - it was complicated enough to make it if-else statement.

Reply:
> I formatted this piece of code a bit differently - it was complicated enough to make it if-else statement.

Looks good @dreamer; it much more readable than the above.